Makefile

```diff
@@ -32,9 +39,15 @@ ta:
 			  $@
 
 .PHONY: clean
+ifneq ($(wildcard $(TA_DEV_KIT_DIR)/host_include/conf.mk),)
 clean:
 	$(q)$(MAKE) -C host/xtest O=$(out-dir)/xtest q=$(q) $@
 	$(q)$(MAKE) -C ta O=$(out-dir)/ta q=$(q) $@
+else
+clean:
+	$(q)echo "TA_DEV_KIT_DIR is not correctly defined - Remove $(out-dir)"
+	$(q)rm -rf $(out-dir)
```

Comment:
Well I'm changing my mind ;o
All things considered, I still prefer `rm -rf $(out-dir)/ta $(out-dir)/xtest` which will always clean stuff properly and does not bring a big risk of accidental deletion since paths end with `ta` and `xtest`.
See, there are even `rm -rf`'s in the Linux makefiles: [here](https://git.kernel.org/cgit/linux/kernel/git/torvalds/linux.git/tree/arch/x86/Makefile?id=refs/tags/v4.3-rc5#n258) ;-)